service/history/transferQueueProcessor.go

```diff
@@ -328,26 +328,26 @@ func (t *transferQueueProcessorImpl) processActivityTask(task *persistence.Trans
 	if err != nil {
 		return err
 	}
+	defer release()
```

Comment:
Can you immediate call 'defer release()' after acquiring the release object?

Reply:
there can be cases when cache is full and context, release being nil, while err not being nil, doing defer release() before error checking can lead to null pointer issue